packages/gatsby-plugin-layout/src/gatsby-node.js

```diff
@@ -0,0 +1,27 @@
+const fs = require(`fs`)
+
+let didRunAlready = false
+
+exports.onPreInit = (_, { component }) => {
+  if (didRunAlready) {
+    throw new Error(`You can only have single instance of gatsby-plugin-layout`)
+  } else if (!component) {
+    throw new Error(`Component option is required`)
```

Comment:
use `src/layouts/index.js` by default to make it drop-in replacement for v1 layouts